pkg/tools/tools.go

```diff
@@ -199,6 +170,102 @@ func ImageImportIntoClusterMulti(ctx context.Context, runtime runtimes.Runtime,
 
 }
 
+func findImages(ctx context.Context, runtime runtimes.Runtime, requestedImages []string) (imagesFromRuntime, imagesFromTar []string, err error) {
+	runtimeImages, err := runtime.GetImages(ctx)
+	if err != nil {
+		log.Errorln("Failed to fetch list of existing images from runtime")
+		return nil, nil, err
+	}
+
+	for _, requestedImage := range requestedImages {
+		if isFile(requestedImage) {
+			imagesFromTar = append(imagesFromTar, requestedImage)
+			log.Debugf("Selected image '%s' is a file", requestedImage)
+			break
+		}
+
+		runtimeImage, found := findRuntimeImage(requestedImage, runtimeImages)
+		if found {
+			imagesFromRuntime = append(imagesFromRuntime, runtimeImage)
+			log.Debugf("Selected image '%s' (found as '%s') in runtime", requestedImage, runtimeImage)
+			break
+		}
+
+		log.Warnf("Image '%s' is not a file and couldn't be found in the container runtime", requestedImage)
+	}
+	return imagesFromRuntime, imagesFromTar, err
+}
+
+func findRuntimeImage(requestedImage string, runtimeImages []string) (string, bool) {
+	canonicalRequestedImage := canonicalImageName(requestedImage)
+
+	for _, runtimeImage := range runtimeImages {
+		if imageNamesEqual(canonicalRequestedImage, runtimeImage) {
+			return runtimeImage, true
+		}
+	}
+	return "", false
+}
+
+func isFile(image string) bool {
+	file, err := os.Stat(image)
+	if err != nil {
+		return false
+	}
+	return !file.IsDir()
+}
+
+// canonicalImageName turns any docker image name in the form `registry/orgName/imageName:versionName`.
+// If `:versionName` is not present, `:latest` is suffixed.
+// If `orgName` is not present, `library` is added.
+// If `registry` is not present, `docker.io` is prefixed.
+func canonicalImageName(image string) string {
+	if !containsVersionPart(image) {
+		image = fmt.Sprintf("%s:latest", image)
+	}
+
+	slashCount := strings.Count(image, "/")
+	switch slashCount {
+	case 0: // zero slashes -> library image tag (e.g. `postgres`)
+		return fmt.Sprintf("docker.io/library/%s", image)
+	case 1: // one slash -> no registry (e.g. `library/postgres`)
+		return fmt.Sprintf("docker.io/%s", image)
+	default: // two slashes or more -> already contains registry name
+		return image
+	}
```

Comment:
Why do we even need to append something here? :thinking: 
In the first iteration, you only stripped out the `docker.io` prefix if it was there.
I guess you could've kept that and just wrap it in `if runtime == runtimes.Docker {...}` to only do this for Docker (though we can change that even later when we implement other runtimes, where this is not true).

Reply:
Addition: I think there is no convention on the number of `/`es, as I just tried with 4 and it was no problem.
As per https://docs.docker.com/engine/reference/commandline/tag/ it seems, the only syntax convention is the `:<tag>` part (and having `/` as a component separator) :thinking: